_integration_tests/test/consumer_api_client/user_tasks/finish_user_task.js

```
@@ -0,0 +1,178 @@
+
+    try {
+      await consumerApiClientService.finishUserTask(consumerContext, processModelKey, correlationId, userTaskId, userTaskResult);
+      should.fail(result, undefined, 'This request should have failed!');
+    } catch (error) {
+      const expectedErrorCode = 404;
+      const expectedErrorMessage = /user task id not found/i
+      should(error.code).match(expectedErrorCode);
+      should(error.message).match(expectedErrorMessage);
+    }
+  });
+
+  // TODO: Bad Path not implemented yet
+  it.skip('should fail to finish the user task, if the given payload is invalid', async () => {
+
+    // TODO: Replace with real values
+    const processModelKey = 'test_consumer_api_user_task_finish';
+    const correlationId = 'correlationId';
+    const userTaskId = 'userTaskId';
+    const userTaskResult = {};
+
+    try {
+      await consumerApiClientService.finishUserTask(consumerContext, processModelKey, correlationId, userTaskId, userTaskResult);
+      should.fail(result, undefined, 'This request should have failed!');
+    } catch (error) {
+      const expectedErrorCode = 400;
+      const expectedErrorMessage = /invalid arguments/i
+      should(error.code).match(expectedErrorCode);
+      should(error.message).match(expectedErrorMessage);
+    }
+  });
+
+  // TODO: Bad Path not implemented yet
+  // TODO: Find a way to simulate a process error
+  it.skip('should fail, if attempting to finish the user task caused an error', async () => {
+
+    // TODO: Replace with real values
+    const processModelKey = 'test_consumer_api_user_task_finish';
+    const correlationId = 'correlationId';
+    const userTaskId = 'invalidUserTaskId';
+    const userTaskResult = {};
+
+    try {
+      await consumerApiClientService.finishUserTask(consumerContext, processModelKey, correlationId, userTaskId, userTaskResult);
+      should.fail(result, undefined, 'This request should have failed!');
+    } catch (error) {
+      const expectedErrorCode = 500;
```

This will most likely be a 4xx - I can't imagine an error scenario that is not caused by invalid user input - this one sounds like an attempt to finish an already finished user task (which would be in contract to the test case right above, but still result in a 4xx)

---

There can be scenarios where this user input is not the cause for the error:

- The userTask you want to finish is malformed (for example if it has no form_fields)
- The Database is offline, and the userTask can't be fetched
- The UserTask is on a Lane that can't be processed by the connected process-engine instance, and no instance that can process the userTask is known or online

---

can we agree on calling tests like this for an unspecified error case something like `should fail on unhandled` or `should fail on unexpected error`?

we should find a wording that makes it easy to understand what kind of error the test is talking about